internal/pkg/nats/jetstream/connection.go

```diff
@@ -0,0 +1,80 @@
+//
+// Copyright (c) 2022 One Track Consulting
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//go:build include_nats_messaging
+
+package jetstream
+
+import (
+	natsMessaging "github.com/edgexfoundry/go-mod-messaging/v2/internal/pkg/nats"
+	"github.com/nats-io/nats.go"
+)
+
+const (
+	autoProvision = "AutoProvision"
+)
+
+// connection mimics the core NATS publish/subscribe API
+// so that NATS and jetstream can use the same client orchestration.
+type connection struct {
+	cfg              natsMessaging.ClientConfig
+	conn             *nats.Conn
+	js               nats.JetStreamContext
+	durableSubscribe bool
+	subOpts          []nats.SubOpt
+	pubOpts          []nats.PubOpt
```

Comment:
I see where this is used, but not where it is set???

Reply:
Set when initializing the jetstream connection, I tried to clean this up a little bit.